GiniVision/Classes/FilePickerManager.swift

```diff
@@ -57,9 +57,14 @@ extension FilePickerManager: UIDocumentPickerDelegate {
             let data = try Data(contentsOf: url)
             url.stopAccessingSecurityScopedResource()
             
-            if let document = GiniVisionDocumentFactory.create(withData: data) {
```

Comment:
I like the factory approach. I understand your consideration and I think you are avoiding a phantom requirement by removing the factory.

For the current state this approach is nice, readable and easy to understand. In case we introduce a new document type with different initializer parameters in the future we can alter this approach.

I would recommend keeping the factory, but moving the factory method to the GiniVisionDocument: `GiniVisionDocument.create(withData: data)`.

What do you think?

Reply:
👍 Thanks for clarifying it: the `GiniImageDocument` will require an initializer with more parameters when you move the meta information manager to it.

Reply:
Apart from what we discussed earlier, I do not think that It would have been a good idea to put `create` factory method inside `GiniVisionDocument` protocol since It should not know about implementation details. I see protocols as 'a set of rules' that do not care about implementation details. What do you think?